src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -50,6 +50,14 @@ public static class ExpressionMetadataProvider
                     // Property/field access is always legal
                     var memberExpression = (MemberExpression)expression.Body;
                     propertyName = memberExpression.Member is PropertyInfo ? memberExpression.Member.Name : null;
+                    if (string.Equals(propertyName, "Model", StringComparison.Ordinal) &&
```

Comment:
As discussed offline, can at least make this more specific and minimize the corner cases: Changed this condition to include `&& memberExpression.Expression.NodeType == ExpressionType.Constant`.